packages/@aws-cdk/aws-stepfunctions/lib/fields.ts

```diff
@@ -219,8 +219,12 @@ function validateJsonPath(path: string) {
     && path !== '$$'
     && !path.startsWith('$$.')
     && !path.startsWith('$[')
+    && !path.startsWith('States.Format')
+    && !path.startsWith('States.StringToJson')
+    && !path.startsWith('States.JsonToString')
+    && !path.startsWith('States.Array')
```

Comment:
I think an array-based condition might be better long term, something like
```suggestion
    && ['Format', 'StringToJson', 'JsonToString', 'Array'].every(fn => !path.startsWith(`States.${fn}`))
```